test/src/nativeTest/kotlin/io/realm/InstrumentedTests.kt

```diff
@@ -80,4 +92,98 @@ class InstrumentedTests {
             assertEquals("Sample", realmModel.`$realm$TableName`)
         }
     }
+
+    // TODO Only run on macOS, filter using https://developer.apple.com/documentation/foundation/nsprocessinfo/3608556-iosapponmac when upgrading to XCode 12
+    @Test
+    fun garbageCollectorShouldFreeNativeResources() {
```

Comment:
I think these memory related test should go to their own file. There is a lot of infrastructure specific to verifying the memory usage, etc. that pollutes the ordinary test cases.

Reply:
Good point, especially for Darwin where some should not be run under iOS and be filtered.